lib/ansible/modules/windows/win_route.ps1

```diff
@@ -0,0 +1,171 @@
+#!powershell
+# This file is part of Ansible
+#
+# Copyright 2016, Daniele Lazzari <lazzari@mailup.com>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+# win_route (Add or remove a network static route)
+
+$params = Parse-Args $args;
```

Comment:
Unfortunately there is no such document for Windows modules yet. However we have updated most of the generic Windows modules to follow the most important rules. So if someone starts from the current modules, they should already be fine for most parts !

We do have the following documents on module development:
- http://docs.ansible.com/ansible/dev_guide/developing_modules_best_practices.html
- http://docs.ansible.com/ansible/dev_guide/developing_modules_checklist.html

You will see that some of my recommendations are documented in that second link.